src/labkey/query/types.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export interface QueryColumn {
```

Comment:
does this interface match anything in the ui-components dir which we can replace with this usage?

Reply:
Yes and no. It does match and could be used but it doesn't buy as much at the moment with `QueryColumn` as it is defined (i.e. as an `Immutable.Record`). I ended up exporting the interface type so that it could be hooked up at a later point if desired.